src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -191,6 +196,40 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod clientMeth
                             writer.Line($");");
                         }
                         break;
+                    case MultipartRequestBody multipartRequestBody:
+                        var multipartContent = new CodeWriterDeclaration("content");
+                        using (WriteValueNullCheck(writer, multipartRequestBody.Value))
+                        {
+                            writer.Line($"var {multipartContent:D} = new MultipartFormDataContent();");
+                            writer.Line($"{multipartContent}.ApplyToRequest({request});");
+                            if (multipartRequestBody.Value.Type.Name == "IEnumerable")
```

Comment:
This condition is interesting, what case is it for?

Reply:
It is for the case when use will [send an `IEnumerable<Stream>`](https://github.com/Azure/autorest.csharp/blob/9d33d8bffd0878a21553ee8f1dfeb142da9fa7fb/test/TestServerProjects/body-formdata/Generated/FormdataRestClient.cs#L230).

Check [this](https://github.com/Azure/autorest.csharp/blob/9d33d8bffd0878a21553ee8f1dfeb142da9fa7fb/test/TestServerProjects/body-formdata/Generated/FormdataRestClient.cs#L192) generated code.